pkgs/build-support/node/fetch-npm-deps/default.nix

```diff
@@ -36,11 +36,12 @@
           '';
         };
 
-        makeTest = { name, src, hash }: testers.invalidateFetcherByDrvHash fetchNpmDeps {
-          inherit name hash;
+        makeTest = { name, src, hash, forceGitDeps ? false }: testers.invalidateFetcherByDrvHash fetchNpmDeps
+          ({
+            inherit name hash;
 
-          src = makeTestSrc { inherit name src; };
-        };
+            src = makeTestSrc { inherit name src; };
+          } // lib.optionalAttrs forceGitDeps { I_LOVE_JS = true; });
```

Comment:
Is this env var only for tests or is it possible it will be needed for actual packages?
If it might need to be used by actual packages, we should probably document it (maybe have it as a normal argument to the fetcher too?)

Reply:
Fixed, except docs.

Reply:
This one looks good now too except for a quick note in the documentation (or it could be in an error diagnostic message tbh)

Reply:
Kind of against noting it in the documentation due to its ability to blow up things (maybe I'm being dumb), but I added it to `buildNpmPackage` + the error message in `prefetch-npm-deps`.

Also: I just realized that we may run into *Git deps with Yarn lockfiles* aaaaaaaaaaaa

Reply:
Noting it just in the error message sounds great to me, and working on it after `buildYarnPackage` also sounds good to me